### PR TITLE
N21 638 error page schoolnumber mismatch

### PR DESCRIPTION
**Pull request by mrikallab**
# Short Description
This PR is about the error page, which will be shown when schoolnumbers mismatch in user migration flow.

## Links to Ticket and related Pull-Requests
https://ticketsystem.dbildungscloud.de/browse/N21-638
This PR should  be merged with Server PR https://github.com/hpi-schul-cloud/schulcloud-server/pull/4006

## Checklist before merging

- [x] QA: In addition to review, the code has been manually tested (if manual testing is possible)
- [x] PO: Any deviation from requirements was agreed with Product-Owner / ticket author / support-team

> Notice: Please keep this Pull-Request as a Draft (or add WIP label), until it is ready to be reviewed

### CODE REVIEW

File: src/router/routes.ts [hunk 1]
```diff
@@ -259,12 +260,20 @@ export const routes: Array<RouteConfig> = [
 			import("@/pages/user-login-migration/UserLoginMigrationError.page.vue"),
 		name: "user-login-migration-error",
 		beforeEnter: validateQueryParameters({
+			errorcode: isString,
 			sourceSystem: isMongoId,
 			targetSystem: isMongoId,
+			sourceSchoolNumber: (value: unknown) =>
+				!isDefined(value) || (isString(value) && /[0-9]{5}/.test(value)),
```
Comment (arnegns) [inline src/router/routes.ts:267]: You could move this function in to a const with a better name and then you would not have code duplication

**Review by arnegns — CHANGES_REQUESTED** [review]

File: src/router/routes.ts (hunk 1 above)
Comment (mrikallab) [reply src/router/routes.ts]: done

File: src/pages/user-login-migration/UserLoginMigrationError.page.vue [hunk 2]
```diff
@@ -4,7 +4,7 @@
 			src="@/assets/img/migration/migration_error.svg"
 			:alt="$t('pages.userMigration.error.img.alt')"
 		/>
-		<h1 class="pl-4 pr-4">
+		<h1 class="pl-4 pr-4" style="white-space: nowrap">
```
Comment (MarvinOehlerkingCap) [inline src/pages/user-login-migration/UserLoginMigrationError.page.vue:7]: no inline style

**Review by MarvinOehlerkingCap — CHANGES_REQUESTED** [review]

File: src/locales/de.json [hunk 3]
```diff
@@ -898,7 +898,8 @@
 	"pages.userMigration.success.description": "Der Umzug des Anmeldesystems von {sourceSystem} zu {targetSystem} wurde erfolgreich abgeschlossen. Dieses Benutzerkonto wurde ausgeloggt und der Login kann jetzt über {targetSystem} erfolgen.",
 	"pages.userMigration.error.img.alt": "Migrationsfehler Grafik",
 	"pages.userMigration.error.title": "Umzug des Anmeldesystems ist fehlgeschlagen",
-	"pages.userMigration.error.description": "Der Umzug des Anmeldesystems von {sourceSystem} zu {targetSystem} ist leider fehlgeschlagen. Um die {instance} mit dem Anmeldesystem {targetSystem} nutzen zu können, bitte <a href=\"{supportLink}\">Kontakt zum Support</a> aufnehmen.",
+	"pages.userMigration.error.description": "Der Umzug des Anmeldesystems von {sourceSystem} zu {targetSystem} ist leider fehlgeschlagen. Um die {instance} mit dem Anmeldesystem {targetSystem} nutzen zu können, bitte <a href=\"{supportLink}\">Kontakt zum Support</a> aufnehmen. ",
+	"pages.userMigration.error.schoolNumberMismatch": "Die Schulnummer <b>{sourceSystem} {sourceSchoolNumber}</b> und die Schulnummer <b>{targetSystem} {targetSchoolNumber}</b> bitte zu der Problemstellung übermitteln.",
```
Comment (MartinSchuhmacher) [inline src/locales/de.json:902]: that sentence does not make sense to me, especially that double "Schulnummer"

File: src/pages/user-login-migration/UserLoginMigrationError.page.vue [hunk 4]
```diff
@@ -20,6 +20,18 @@
 					})
 				"
 			></p>
+			<p
+				data-testId="text-schoolNumberMismatch"
```
Comment (MartinSchuhmacher) [inline src/pages/user-login-migration/UserLoginMigrationError.page.vue:24]: why mixing "-" and camelCase here? Please choose one way

File: src/router/routes.ts [hunk 5]
```diff
@@ -261,10 +263,16 @@ export const routes: Array<RouteConfig> = [
 		beforeEnter: validateQueryParameters({
 			sourceSystem: isMongoId,
 			targetSystem: isMongoId,
+			sourceSchoolNumber: (value: unknown) =>
+				!isDefined(value) || isOfficialSchoolNumber(value),
```
Comment (MartinSchuhmacher) [inline src/router/routes.ts:267]: so validating this two number means its either "not defined" OR "an official school number"?
validating something via checking if its "not defined" sounds strange to me

**Review by MartinSchuhmacher — CHANGES_REQUESTED** [review]

File: src/locales/de.json (hunk 3 above)
Comment (mrikallab) [reply src/locales/de.json:902]: Hi Martin, 
this sentence was discussed with UI/UX, it was agreed on february, 21, 2023.

**Review by MarvinOehlerkingCap — APPROVED** [review]

File: src/pages/user-login-migration/UserLoginMigrationError.page.vue [hunk 6]
```diff
@@ -21,7 +21,7 @@
 				"
 			></p>
 			<p
-				data-testId="text-schoolNumberMismatch"
+				data-testId="text-schoolnumbermismatch"
```
Comment (MartinSchuhmacher) [inline src/pages/user-login-migration/UserLoginMigrationError.page.vue:24]:
```suggestion
				data-testId="text-schoolnumber-mismatch"
```

File: src/pages/user-login-migration/UserLoginMigrationError.page.unit.ts [hunk 7]
```diff
@@ -117,7 +117,7 @@ describe("UserLoginMigrationError", () => {
 				});
 
 				const schoolNumberMismatchText: string = wrapper
-					.find("[data-testId=text-schoolNumberMismatch]")
+					.find("[data-testId=text-schoolnumbermismatch]")
```
Comment (MartinSchuhmacher) [inline src/pages/user-login-migration/UserLoginMigrationError.page.unit.ts:120]:
```suggestion
					.find("[data-testId=text-schoolnumber-mismatch]")
```

File: src/router/routes.ts (hunk 5 above)
Comment (MartinSchuhmacher) [reply src/router/routes.ts:267]: understood 👍

**Review by MartinSchuhmacher — APPROVED** [review]

**Review by MartinSchuhmacher — DISMISSED** [review]

**Review by arnegns — DISMISSED** [review]

**Pull request merged**